pkg/daemon/daemon.go

```diff
@@ -282,11 +269,31 @@ func (dn *Daemon) Run(stopCh <-chan struct{}, exitCh <-chan error) error {
 	// handleNodeUpdate(). If a failure happens there, it writes to the channel.
 	// The HealthzMonitor goroutine also writes to this channel if the threshold
 	// is reached.
-	err = <-exitCh
+	err := <-exitCh
 
 	return dn.nodeWriter.SetUpdateDegradedIgnoreErr(err, dn.kubeClient.CoreV1().Nodes(), dn.name)
 }
 
+func (dn *Daemon) runLoginMonitor(stopCh <-chan struct{}, exitCh chan<- error) {
+	sessionNewCh := dn.loginClient.Subscribe("SessionNew")
+	for {
+		select {
+		case <-stopCh:
+			return
+		case msg := <-sessionNewCh:
+			glog.Infof("Detected a new logged in session: %v", msg)
+			glog.Infof("Login session taints node")
+			err := replaceFileContentsAtomically(pathSSHTaint, []byte("tainted"))
+			if err != nil {
+				glog.Errorf("Writing taint file: %v", err)
+			}
+			if err := dn.applySSHTaint(); err != nil {
+				glog.Errorf("Error during taint: cannot apply taint due to: %v", err)
+			}
```

Comment:
We'll need to remember to write any errors here to the `exitCh` once we enable this. Might be worth adding a comment about that somewhere in there.